internal/service/streamprivatelinkendpoint/resource_test.go

```diff
@@ -1,34 +1,242 @@
+	t.Helper()
+
+	var (
+		projectID    = acc.ProjectIDExecution(t)
+		provider     = "AWS"
+		region       = "us-east-1"
+		vendor       = "CONFLUENT"
+		awsAccountID = os.Getenv("AWS_ACCOUNT_ID")
+	)
+
+	return &resource.TestCase{
+		PreCheck:                 func() { acc.PreCheckBasic(t) },
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		CheckDestroy:             checkDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: configConfluentDedicatedCluster(provider, region, awsAccountID) + configBasic(projectID, provider, region, vendor, false),
+				Check:  checksStreamPrivatelinkEndpoint(projectID, provider, region, vendor, false),
+			},
+			{
+				Config:      configBasic(projectID, provider, region, vendor, true),
+				ExpectError: regexp.MustCompile(`Operation not supported`),
+			},
+		},
+	}
+}
+
+func configBasic(projectID, provider, region, vendor string, withDNSSubdomains bool) string {
+	dnsSubDomainConfig := ""
+	if withDNSSubdomains {
+		dnsSubDomainConfig = `dns_sub_domain = local.dns_sub_domain_entries`
+	}
+
+	return fmt.Sprintf(`
+	locals {
+		dns_sub_domain_entries = [
+    		for zone in confluent_network.private-link.zones :
+    		"${zone}.${confluent_network.private-link.dns_domain}"
+  		]
+	}	
+
+	resource "mongodbatlas_stream_privatelink_endpoint" "test" {
+		project_id          = %[1]q
+		dns_domain          = confluent_network.private-link.dns_domain
+		provider_name       = %[2]q
+		region              = %[3]q
+		vendor              = %[4]q
+		service_endpoint_id = confluent_network.private-link.aws[0].private_link_endpoint_service
+		%[5]s
+	}
+
+	data "mongodbatlas_stream_privatelink_endpoint" "singular-datasource-test" {
+		project_id = %[1]q
+		id         = mongodbatlas_stream_privatelink_endpoint.test.id
+	}
+
+	data "mongodbatlas_stream_privatelink_endpoints" "plural-datasource-test" {
+		project_id = %[1]q
+	}`, projectID, provider, region, vendor, dnsSubDomainConfig)
+}
+
+func configConfluentDedicatedCluster(provider, region, awsAccountID string) string {
+	return fmt.Sprintf(`
+	%[1]s
+
+	data "confluent_environment" "default_environment" {
+  		display_name = "default"
+	}
+
+	resource "confluent_network" "private-link" {
+		display_name     = "terraform-test-private-link-network"
+		cloud            = %[2]q
+		region           = %[3]q
+		connection_types = ["PRIVATELINK"]
+		zones            = ["use1-az1", "use1-az4", "use1-az6"]
+		environment {
+			id = data.confluent_environment.default_environment.id
+		}
+		dns_config {
+			resolution = "PRIVATE"
+		}
+	}
+
+	resource "confluent_private_link_access" "aws" {
+		display_name = "terraform-test-aws-private-link-access"
+		aws {
+			account = %[4]q
+		}
+		environment {
+			id = data.confluent_environment.default_environment.id
+		}
+		network {
+			id = confluent_network.private-link.id
+		}
+	}
+
+	resource "confluent_kafka_cluster" "dedicated" {
+		display_name = "terraform-test-cluster"
```

Comment:
Does this name need to be randomized? Thinking about parallel runs

Reply:
We want to avoid as much as possible parallel runs, Confluent resources have a significant cost. It is intentional to have a hardcoded name. Will try to avoid depending on the cluster (see other comment reply)